tests/integ/test_packaging.py

```diff
@@ -539,3 +575,258 @@ def plus_one_month(x):
+    """
+    test_files = TestFiles(resources_path)
+
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        with patch(
+            "snowflake.snowpark.session.Session._load_unsupported_packages_from_stage",
+            side_effect=Exception("This function does not work"),
+        ):
+            session.add_requirements(
+                test_files.test_unsupported_requirements_file,
+                persist_path=temporary_stage,
+            )
+
+    assert session.get_packages().keys() == {"matplotlib", "numpy", "pyyaml", "scipy"}
+
+    udf_name = Utils.random_name_for_temp_object(TempObjectType.FUNCTION)
+
+    @udf(name=udf_name)
+    def get_numpy_pandas_version() -> str:
+        import skfuzzy as fuzz
+
+        return fuzz.__version__
+
+    Utils.check_answer(session.sql(f"select {udf_name}()"), [Row("0.4.2")])
+
+
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC,
+    reason="Subprocess calls are not allowed within stored procedures",
+)
+def test_add_requirements_unsupported_with_persist_path(
+    session, resources_path, temporary_stage
+):
+    """
+    Assert that if a persist_path is mentioned, the zipped packages file and a metadata file are present at this
+    remote stage path. Also, subsequent attempts to add the same requirements file should result in the zip file
+    being directly imported from persist_path (i.e. no pip install, no native package dependency detection, etc).
+    We test this by patching the `_upload_unsupported_packages` function to throw an Exception.
+
+    Finally, assert that adding a new unsupported package results in a new environment signature and zip file (i.e.
+    two environments should be present on the stage).
+    """
+    test_files = TestFiles(resources_path)
+
+    # Prove that patching _upload_unsupported_packages leads to failure
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        with patch(
+            "snowflake.snowpark.session.Session._upload_unsupported_packages",
+            side_effect=Exception("This function should not have been called"),
+        ):
+            with pytest.raises(Exception) as ex_info:
+                session.add_requirements(
+                    test_files.test_unsupported_requirements_file,
+                    persist_path=temporary_stage,
+                )
+            assert "This function should not have been called" in str(ex_info)
+
+    session.clear_imports()
+    session.clear_packages()
+
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        session.add_requirements(
+            test_files.test_unsupported_requirements_file, persist_path=temporary_stage
+        )
+        # Once scikit-fuzzy is supported, this test will break; change the test to a different unsupported module
+
+    environment_hash = "43c5b9d5af61620d2efe4e6fafce11901830f080"
+    zip_file = f"{IMPLICIT_ZIP_FILE_NAME}_{environment_hash}.zip"
```

Comment:
can you use the the same hash function used in the library to generate this hash?